### PR TITLE
Advertise make compiler in configure

### LINKED ISSUE #4450: [opam 2.1~beta4] ./configure requires an OCaml compiler but a vendored OCaml compiler will be compiled anyway
Here is my setup:
```
$ cat /etc/os-release | grep ID=
ID=ubuntu
VERSION_ID="18.04"
$ git clone git://github.com/ocaml/opam.git
$ cd opam
$ ./configure
./configure
checking for ocamlc... no
checking for ocaml... no
checking for ocamldep... no
checking for ocamlmktop... no
checking for ocamlmklib... no
checking for ocamldoc... no
checking for ocamlbuild... no
configure: error: You must install the OCaml compiler
$ sudo apt install ocaml-nox
$ ocaml -vnum
4.05.0
$ ./configure
$ make lib-ext
$ make
[...]
make[4]: Entering directory '/home/kit_ty_kate/opam/src_ext/secondary/ocaml-4.11.1/stdlib'
../boot/ocamlrun ../ocamlc -strict-sequence -absname -w +a-4-9-41-42-44-45-48 -g -warn-error A -bin-annot -nostdlib -safe-string -strict-formats  -nopervasives -c camlinternalFormatBasics.mli
../boot/ocamlrun ../ocamlc -strict-sequence -absname -w +a-4-9-41-42-44-45-48 -g -warn-error A -bin-annot -nostdlib -safe-string -strict-formats  -nopervasives -c camlinternalFormatBasics.ml
```
The last bit indicates that a vendored OCaml compiler is being compiled. However the configure script demands to have an OCaml compiler installed. To me, this seems like extra steps that's not needed.

### PR DESCRIPTION
**Pull request by dra27**
Fixes #4450

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -16,7 +16,7 @@ AS_IF([ test -x bootstrap/ocaml/bin/ocamlc -o -x bootstrap/ocaml/bin/ocamlopt ],
 
 AC_PROG_OCAML
 if test "x$OCAMLC" = "xno"; then
-   AC_MSG_ERROR([You must install the OCaml compiler])
+   AC_MSG_ERROR([You must install the OCaml compiler (or run: make compiler)])
```
Comment (kit-ty-kate) [inline configure.ac:19]:
```suggestion
   AC_MSG_ERROR([You must install an OCaml compiler >= 4.07 (or run: make compiler)])
```

Comment (kit-ty-kate) [reply configure.ac:19]:
```suggestion
   AC_MSG_ERROR([You must install the OCaml compiler >= 4.02.3, ideally >= 4.07.0, no biggy if not (or run: make compiler)])
```
or something like that ^^"

Comment (dra27) [reply configure.ac:19]: Meh, it’s another message to maintain a version number in...

Comment (dra27) [reply configure.ac]: Updated to include the minimum version for building opam, as that's already in `configure` (if a little catch-22-ish for Windows!)

**Review by rjbou — APPROVED** [review]

**Comment by rjbou** [comment]
@dra27 good to go ?

**Pull request merged**